pkg/services/serviceaccounts/api/api.go

```diff
@@ -42,7 +42,7 @@ type service interface {
 	MigrateApiKeysToServiceAccounts(ctx context.Context, orgID int64) (*serviceaccounts.MigrationResult, error)
 	MigrateApiKey(ctx context.Context, orgID int64, keyId int64) error
 	// Service account tokens
-	AddServiceAccountToken(ctx context.Context, serviceAccountID int64, cmd *serviceaccounts.AddServiceAccountTokenCommand) (*apikey.APIKey, error)
+	AddServiceAccountToken(ctx context.Context, serviceAccountID int64, cmd *serviceaccounts.AddServiceAccountTokenCommand) (*apikey.APIKey, string, error)
```

Comment:
wanted to change the interface from

```
AddServiceAccountToken(ctx context.Context, serviceAccountID int64, cmd *serviceaccounts.AddServiceAccountTokenCommand) (*apikey.APIKey, error)
```

```
	AddServiceAccountToken(ctx context.Context, serviceAccountID int64, cmd *serviceaccounts.AddServiceAccountTokenCommand) (*apikey.APIKey, string, error)
```

Because key generation happens outside, is this because we want to keep track of the services that create the keys?

Perhaps, we want to add that as a parameter and return the client secret